gpjax/dataset.py

```diff
@@ -31,8 +30,8 @@ class Dataset(Pytree):
         y (Optional[Float[Array, "N Q"]]): Output data.
     """
 
-    X: Optional[Float[Array, "N D"]] = None
-    y: Optional[Float[Array, "N Q"]] = None
+    X: Optional[Union[Num[Array, "N D"], Num[Array, "..."]]] = None
+    y: Optional[Union[Float[Array, "N Q"], Float[Array, "..."]]] = None
```

Comment:
should this also be
```suggestion
    y: Optional[Union[Num[Array, "N Q"], Num[Array, "..."]]] = None
```
for e.g. classification?